paper/src/main/java/com/devotedmc/ExilePearl/listener/PlayerListener.java

```diff
@@ -572,7 +573,13 @@ else if(event.getAction() == InventoryAction.MOVE_TO_OTHER_INVENTORY) {
 		}
 		else if(event.getAction() == InventoryAction.HOTBAR_SWAP) {
 			PlayerInventory playerInventory = event.getWhoClicked().getInventory();
-			ExilePearl pearl = pearlApi.getPearlFromItemStack(playerInventory.getItem(event.getHotbarButton()));
+			ItemStack itemStack = null;
+			try {
+				itemStack = playerInventory.getItem(event.getHotbarButton());
+			} catch (ArrayIndexOutOfBoundsException exception) {
+				itemStack = playerInventory.getItem(EquipmentSlot.OFF_HAND);
+			}
+			ExilePearl pearl = pearlApi.getPearlFromItemStack(itemStack);
```

Comment:
Per my test event.getHotbarButton() returns -1 for the case with off-hand.
Therefore I propose to check if it is negative or not instead of catching exception (raising exception has negative influence on performance):
![image](https://user-images.githubusercontent.com/5383652/178086180-e861e7a2-55da-4f85-8745-95f1ead8250f.png)